### ISSUE TITLE
Quarry particle effect bad texture

### ISSUE THREAD

**Issue by S-S-X** [mt-mods/technic#210]
https://github.com/mt-mods/technic/blob/aaf4f4768e6258203292164407e3e06fa8da5267/technic/machines/HV/quarry.lua#L409-L417

Texture wont work with `minetestserver`, it will only work with `minetest --server`. That's because it uses ContentDB texture and `minetestserver` is built without those.
Simply switch to some texture that is always available, also size most probably must be changed when updating texture so test first.

Added here: https://github.com/mt-mods/technic/pull/179
Comments here: https://github.com/mt-mods/technic/commit/becb2468b747cdc5b46c23737fd28b8d101057cf#commitcomment-51037814 (also about particle sizing)

**Comment by S-S-X** [comment]
Easy option would be to just use texture that's commented out, set size to 10 and it should look like this: https://www.youtube.com/watch?v=PCRGBbp8tJU
Not 10% sure if size 10 is correct... should be very close anyway.

**Comment by OgelGames** [comment]
Could also just add a texture specifically for it...

Also, wouldn't it be possible to do the same thing with a few entities, similar to how worldedit does it? Even if extra entities are used to make sure they are visible, it should be possible to do with less than 6 or 7 entities if my thinking is correct...

https://github.com/Uberi/Minetest-WorldEdit/blob/master/worldedit_commands/mark.lua

**Comment by S-S-X** [comment]
> Even if extra entities are used to make sure they are visible, it should be possible to do with less than 6 or 7 entities if my thinking is correct...

Yes, reason why I specifically approached with particle effect is simply that it way cheaper for server. Particles being fire and forget type (server simply sends parameter to client and forgets it) there's no server side management for those unlike entities that have are fully managed by server.

Another reason was that particles can be sent only to selected players, there only player who actually does hit quarry will know about involved particles. None of any other connected clients have any idea about those particles and therefore particle effect does not affect anyone else in any way.

**Comment by OgelGames** [comment]
Yes, I did read the PR, and I agree with your reasoning, but it wouldn't be much for something that would look a lot better (and be easier to use).

> offtopic side note: per player particle effects with some limits could be okayish way to do that digiline particle effect debugger

👍

**Comment by BuckarooBanzay** [comment]
> Easy option would be to just use texture that's commented out, set size to 10 and it should look like this: https://www.youtube.com/watch?v=PCRGBbp8tJU
> Not 100% sure if size 10 is correct... should be very close anyway.

looks good in the video :+1:

**Comment by OgelGames** [comment]
![quarry_visuals](https://user-images.githubusercontent.com/48543043/132719314-efefc192-02e2-4740-b85b-44569b5b16b0.png)

**Comment by S-S-X** [comment]
Entities looks nice, a lot better than what particle effects are capable of.

However I would keep entities away because those have always introduced more problems because of very long actively managed lifecycles and relatively speaking extremely complex server side management.

In my opinion it would be better to remove digging area visualization completely if it seems that simple visualization for player who activates visual aid is not good enough. In that case it would be better to wait until engine offers proper way to do entity like 3d visuals client side.

Entities are visible for everyone (which in my opinion is very bad for this use case) and have long history of problems with other mods, I'd say better to not use entities because in my opinion even while it looks nice it brings more problems for very little good. Either stick with particles (where only expense is sending parameters to client) or remove feature and wait for proper engine support.

**Comment by OgelGames** [comment]
Meanwhile jumpdrives, protection blocks and postool use entities in the exact same way, without anyone suggesting they should use particles...

**Comment by S-S-X** [comment]
> Meanwhile jumpdrives, protection blocks and postool use entities in the exact same way, without anyone suggesting they should use particles...

It is not really good to defend these things by saying that everything else is already bad so why not add more of that...

There's a lot of people suggesting that they should use particle like system (see minetest issue tracker for requests and reasoning).

Problem is that nothing else was available back then and situation is still same: there is no good way to do these things.

> (sorry if I'm a bit persistent, but I've disliked the particles since they were added, I just hadn't said anything)

You're not only one who dislikes particles, this is not about defending particles. This is about not adding things that in my opinion should not be added. Particles AFAIK is currently only possible way to do that while guaranteeing that it does not affect anything else, not in game and not in code.

If you really do hate particles that much then I do suggest again: remove feature completely and wait for proper engine support (after all I added this just because one player told it would be good). Btw, everyone agreed that visuals should be hidden from everyone else (same should be done with all the mods you just listed but is not really possible for all those).

**Comment by S-S-X** [comment]
To be clear I'll repeat once again: I do think that particle system is very bad for this. It is ONLY available system that does not affect anything else.

**Comment by OgelGames** [comment]
> It is not really good to defend these things by saying that everything else is already bad so why not add more of that...

I'm not, I'm saying they are good, and that we should use the same perfectly acceptable solution until something better is possible. (per-player entities or 3d particles)

> There's a lot of people suggesting that they should use particle like system (see minetest issue tracker for requests and reasoning).

Can you link those? I could only find this: https://github.com/minetest/minetest/issues/8045

**Comment by Klaranth** [comment]
Learned something new today, thanks to @OgelGames . Did not know hitting the quarry gave information.  But frankly, I would prefer it responds the same way a JD does. Just giving the four corners of what it will mine.

**Comment by S-S-X** [comment]
> Learned something new today, thanks to @OgelGames . Did not know hitting the quarry gave information. But frankly, I would prefer it responds the same way a JD does. Just giving the four corners of what it will mine.

Yeah, I would also prefer that. What I do want is things that are happening under the hood in server and clients. Now it gone to this I really regret ever adding that feature...
